app/assets/javascripts/discourse/app/components/second-factor-input.gjs

```diff
@@ -1,7 +1,19 @@
 import Component from "@glimmer/component";
+import { schedule } from "@ember/runloop";
 import { SECOND_FACTOR_METHODS } from "discourse/models/user";
 
 export default class SecondFactorInput extends Component {
+  constructor() {
+    super(...arguments);
+    this._focusInput();
+  }
+
+  _focusInput() {
+    schedule("afterRender", () => {
+      document.querySelector(".second-factor-token-input").focus();
+    });
+  }
+
```

Comment:
I would just use didInsert for all this logic, and not even sure you need schedule in this case and you would avoid the document.querySelector which is not a good practice if we can avoid it